### PR TITLE
Added functionality to convert major, minor, patch to version

### PR DESCRIPTION
**Pull request by LaurentGoderre**
In cases where the binary classifier is able to pick up individually on a major, minor, and patch version fields, but not a semantic version, then the `version` parameter used will be filled in with `major.minor.patch` in the package metadata.

### CODE REVIEW

File: syft/pkg/cataloger/binary/classifier_test.go
Comment (wagoodman) [inline syft/pkg/cataloger/binary/classifier_test.go:1]: it looks like this test has been added to test the effect of `version` being filled out somewhere, but it hasn't been explicitly tested yet. Maybe this is partially completed?

Comment (LaurentGoderre) [reply syft/pkg/cataloger/binary/classifier_test.go:1]: I found that the tests don't actually test whether the version is right but whether it is detected so I would to add new test for both the current approach and the new one.

Comment (LaurentGoderre) [reply syft/pkg/cataloger/binary/classifier_test.go:1]: Added the tests

**Review by spiffcs — APPROVED** [review]

**Comment by spiffcs** [comment]
Note this is usful for binaries like [haxe](https://github.com/HaxeFoundation/haxe) which incorporate the major minor patch versions as metadata in the binary and not as a specific semver string.

**Pull request merged**